dags/operators/dataproc_hadoop_with_aws.py

```diff
@@ -0,0 +1,46 @@
+from airflow.contrib.hooks.aws_hook import AwsHook
+from airflow.contrib.hooks.gcp_dataproc_hook import DataProcHook
+from airflow.contrib.operators.dataproc_operator import DataProcHadoopOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class DataProcHadoopOperatorWithAws(DataProcHadoopOperator):
+    @apply_defaults
+    def __init__(self, aws_conn_id=None, **kwargs):
+        super(DataProcHadoopOperatorWithAws, self).__init__(**kwargs)
+        self.aws_conn_id = aws_conn_id
+
+    def execute(self, context):
+        # add s3 credentials to the job without exposing them in the airflow UI
+        properties = {
+            "fs.s3a." + key: value
+            for key, value in zip(
+                ("access.key", "secret.key", "session.token"),
+                AwsHook(aws_conn_id=self.aws_conn_id).get_credentials(),
+            )
+            if value is not None
+        }
+        if self.dataproc_properties is not None:
+            properties.update(self.dataproc_properties)
+
+        # fork super().execute to reference combined properties. forked from:
```

Comment:
Would this work if `self.dataproc_properties` were set via `self.dataproc_properties.update(properties)`, followed by `super().execute()`?

Reply:
I expect `self.dataproc_properties.update(properties)` would expose the credentials in the airflow UI, but I'll check.

Reply:
I see, so the class/instance variables associated with the operator are displayed in the Airflow UI, so keeping the properties as a local variable is the workaround?

Reply:
yes

Reply:
also, it appears that the UI is reading the values from class instances before `execute` gets called, so that it doesn't show the credentials, however I don't want to rely on that behavior.